hamp_radar/iquick.py

```diff
@@ -116,15 +244,41 @@ def compact_geometry(
 
 
 def extract_raw_arrays(data, mmbgs: Iterable[MultiMainBlockGeometry]):
+    """
+    Generator function to extract arrays from 'data' based on the
+    geometry given by the iterator over MultiMainBlockGeometry instances.
+
+    Iterating over a list of this generator yields a tuple for the subblocks
+    across all the MultiMainBlockGeometry instances sequentially.
+
+    Optimisation uses NumPy library as_strided function to create a view of the
+    original data array interpreted with different shape and strides. Shape will
+    have (nrows, ncols) = (mmbg.count, block.size). Stride will be 1 unless
+    mmbg.count > 1, in which case mmbg.step is used to advance to the next
+    required subblock (skipping past other subblocks with different tags)
```

Comment:
I'd also skip this, as this is an implementation detail and not super helpful for external users. Also I think especially with the use of keyword arguments, the code should now be pretty much equivalent to the text.
```suggestion
```